Python/Product/PythonTools/PythonTools/Intellisense/ProjectAnalyzer.cs

```diff
@@ -188,19 +186,17 @@ public sealed class VsProjectAnalyzer : IDisposable {
             entry.Properties[_pathInZipFile] = value;
         }
 
-        private void OnModulesChanged(object sender, EventArgs e) {
+        private async void OnModulesChanged(object sender, EventArgs e) {
             Debug.Assert(_pyAnalyzer != null, "Should not have null _pyAnalyzer here");
             if (_pyAnalyzer == null) {
                 return;
             }
 
-            lock (_contentsLock) {
-                _pyAnalyzer.ReloadModulesAsync().WaitAndUnwrapExceptions();
+            await _pyAnalyzer.ReloadModulesAsync();

```

Comment:
We had to deal with that possibility anyway (and we don't do a good job :( ) as it was being raised from a background thread already. Fairly sure we aren't any less safe now than we were before, unless we hit a crash where previously we'd have hit the deadlock first.